src/Fees.sol

```diff
@@ -1,58 +1,69 @@
+    uint256 constant SYBIL_FEE = FIL_TO_ATTO_FIL / 10;
+
+    // 2 USD/Tib/month is the current reward earned by Storage Providers
+    uint256 constant MONTHLY_TIB_STORAGE_REWARD_USD = 2;
+    // 1% of reward per period
+    uint256 constant PROOF_FEE_PERCENTAGE = 1;
+    // 4% of reward per period for gas limit left bound
+    uint256 constant GAS_LIMIT_LEFT_PERCENTAGE = 4;
+    // 5% of reward per period for gas limit right bound
+    uint256 constant GAS_LIMIT_RIGHT_PERCENTAGE = 5;
+    uint256 constant USD_DECIMALS = 1e18;
+
+    // 1 TiB in bytes (2^40)
+    uint256 constant TIB_IN_BYTES = 2 ** 40;
+    // Number of epochs per month (30 days * 2880 epochs per day)
+    uint256 constant EPOCHS_PER_MONTH = 86400;
+
+    /// @return proof fee in AttoFIL
+    function proofFeeWithGasFeeBound(
+        uint256 estimatedGasFee, // in AttoFIL
+        uint64 filUsdPrice,
+        int32 filUsdPriceExpo,
+        uint256 rawSize,
+        uint256 nProofEpochs
+    ) internal pure returns (uint256) {
+        require(estimatedGasFee > 0, "failed to validate: estimated gas fee must be greater than 0");
+        require(filUsdPrice > 0, "failed to validate: AttoFIL price must be greater than 0");
+        require(rawSize > 0, "failed to validate: raw size must be greater than 0");
+
+        // Calculate reward per epoch per byte (in AttoFIL)
+        uint256 rewardPerEpochPerByte;
+        if (filUsdPriceExpo >= 0) {
+            rewardPerEpochPerByte = (MONTHLY_TIB_STORAGE_REWARD_USD * FIL_TO_ATTO_FIL) / (TIB_IN_BYTES * EPOCHS_PER_MONTH * filUsdPrice * (10 ** uint32(filUsdPriceExpo)));
+        } else {
+            rewardPerEpochPerByte = (MONTHLY_TIB_STORAGE_REWARD_USD * FIL_TO_ATTO_FIL * (10 ** uint32(-filUsdPriceExpo))) /
+            (TIB_IN_BYTES * EPOCHS_PER_MONTH * filUsdPrice);
+        }
+
+        // Calculate total reward for the proving period
+        uint256 rewardPerPeriod = rewardPerEpochPerByte * nProofEpochs * rawSize;
+
+        // Calculate gas limits
+        uint256 gasLimitRight = (rewardPerPeriod * GAS_LIMIT_RIGHT_PERCENTAGE) / 100;
+        uint256 gasLimitLeft = (rewardPerPeriod * GAS_LIMIT_LEFT_PERCENTAGE) / 100;
 
-    // Returns the proof fee for a given challenge count
-    function proofFee(uint256 challengeCount) internal view returns (uint256) {
-        uint256 gasPrice;
-        if (block.basefee > ONE_NANO_FIL) {
-            gasPrice = block.basefee;
+        if (estimatedGasFee >= gasLimitRight) {
+            return 0; // No proof fee if gas fee is above right limit
+        } else if (estimatedGasFee >= gasLimitLeft) {
+            return gasLimitRight - estimatedGasFee; // Partial discount on proof fee
         } else {
-            gasPrice = ONE_NANO_FIL;
+            return (rewardPerPeriod * PROOF_FEE_PERCENTAGE) / 100;
         }
-        uint256 numerator = PROOF_GAS_FLOOR * challengeCount * gasPrice;
-        uint256 denominator = ONE_PERCENT_DENOM;
-        return numerator / denominator;
     }
 
```

Comment:
Since the last iteration this has become so much cleaner thanks everybody for simplifying.